sniffles/snifflesconfig.py

```diff
@@ -459,7 +466,20 @@ def parse_opt_arg(self, opt, arg):
             self.tcp_ack = True
         # Set percentage of background traffics to be added
         elif opt == "-B":
-            self.background_traffic = int(arg)
+            args = re.split('[:]+', arg)
+            if len(args) == 1:
+                self.background_traffic = int(args[0])
+            else:
+               args = re.split('[,]+', arg)
+               for i in range(0, len(args)):
+                   args_dist = re.split('[:]+', args[i])
+                   if len(args_dist) == 1:
+                       self.usage()
+                   else:
+                       self.background_traffic += int(args_dist[1])
```

Comment:
What I had intended was that you create the BackgroundTrafficRule here.  so between line 473 and 474 you would have something like:
  self.background_traffic_rule = BackgroundTrafficRule()
then here, you would have:
self.background_traffic_rule.setDistribution(arg_dist[0], int(arg_dist[1]))

or something along those lines.
Thus, the lines from sniffles.py (110 - 120) could be removed or moved to here.  Does that make sense?  Unless there is something I am missing here.
Also note, BackgroundTrafficRule has a getDistribution to get the distribution in whole.